library/src/google-maps-provider.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {useState, useEffect, PropsWithChildren} from 'react';
+
+// https://developers.google.com/maps/documentation/javascript/url-params
+export interface GoogleMapsAPIUrlParameters {
+  libraries?: string[];
+  language?: string;
+  region?: string;
+  version?: string;
+  authReferrerPolicy?: string;
+}
+
+export interface GoogleMapsProviderProps extends GoogleMapsAPIUrlParameters {
+  googleMapsAPIKey: string;
```

Comment:
I think the api key should also be in the interface `GoogleMapsAPIUrlParameters`
as it is one of the required parameters: https://developers.google.com/maps/documentation/javascript/url-params#required_parameters

Reply:
But the parameter in the interface is called `key`, not `googleMapsAPIKey`. Do you want to rename it to `key`? Or how would you handle this?

Reply:
No, I think, I would leave it as `googleMapsAPIKey` because it is still up us how to name things, and it sounds a bit clearer than `key` in this case. But, still considering the name of the interface `GoogleMapsAPIUrlParameters`, I would say that it belongs to the parameter as in the documentation (IMO). Just feels more natural as it is similar to the (without React) setup.

For the naming: We could also think of renaming the `MapsProvider` to `GoogleMapsProvider` and then we could just say `key`here as it would be clear. But I am not sure, if the renaming would cause more trouble or is actually a good idea :D

Reply:
Ok yes, I prefer `googleMapsAPIKey` as well. I just wanted to point out, that the `GoogleMapsAPIUrlParameters` wouldn't be the same as in Google's Documentation, because we call it `googleMapsAPIKey` instead of `key`.

The provider is still named `GoogleMapsProvider`. 🤔